band/scheduler/graph_scheduler.h

```diff
@@ -0,0 +1,21 @@
+#ifndef BAND_SCHEDULER_GRAPH_SCHEDULER_H_
+#define BAND_SCHEDULER_GRAPH_SCHEDULER_H_
+
+#include "band/context.h"
+#include "band/graph/graph.h"
+
+namespace band {
+
+class IGraphScheduler {
```

Comment:
I guess we somehow need to refactor `Job` to embed a graph instead of the single model.
This separate scheduling interface for the graph would require a massive amount of change in the planner.

Reply:
Yes, that is the biggest concern that I had while writing code about this along with the #236.
I was thinking about making `Task` a representation that contains the whole information for the execution of a frame, and `Job` remains as an execution unit of a `Subgraph`. So, `GraphScheduler` transforms `Graph` to `Task` containing required `Job`s.

Reply:
Also, it is scary that a `Graph` should contain a general computation.

Reply:
That sounds good to me. My major concern was the separation of scheduling interfaces.

Instead of the large 'Job', we could make ‘Task’ to hold dependencies between Job (name is a bit ambiguous though) and some other info.
And let scheduler interface to get a set of 'Job' and 'Task's to cover both model and graph schedulers.

We may need a better name to both 'Job' and 'Task'

Reply:
I think the translation from Graph to Task/Job is the Engine's responsibility not Planner or Scheduler.